packages/compartment-mapper/test/project-fixture.js

```diff
@@ -0,0 +1,301 @@
+      wait = async () => {};
+    }
+
+    await wait();
+
+    const packageName = nodePath.dirname(filepath);
+    const dependencies = graph[packageName] || [];
+
+    /** @type {PackageDescriptor} */
+    const packageDescriptor = {
+      name: packageName,
+      version: '1.0.0',
+      dependencies: fromEntries(
+        dependencies.map(dependencyName => [dependencyName, '1.0.0']),
+      ),
+    };
+
+    return Buffer.from(JSON.stringify(packageDescriptor));
+  };
+
+/**
+ * Creates `ReadPowers` for use with a {@link ProjectFixture}
+ *
+ * @param {ProjectFixture} fixture
+ * @param {MakeProjectFixtureReadPowersOptions} [options]
+ */
+export const makeProjectFixtureReadPowers = (
+  fixture,
+  {
+    fs = nodeFs,
+    url = nodeUrl,
+    crypto = nodeCrypto,
+    path = nodePath,
+    ...otherOptions
+  } = {},
+) => {
+  return {
+    ...makeReadPowers({ fs, url, crypto, path }),
+    maybeRead: makeMaybeReadProjectFixture(fixture, otherOptions),
+  };
+};
+
+/**
+ * If `value` is an object and has a `null` prototype, re-create it as a "plain"
+ * object to suppress the annoying `[Object: null prototype]` in `inspect()`.
+ *
+ * In the above case, returns a shallow copy of `value`. Any nested object fulfilling the same condition will also be converted and become a shallow copy.
+ *
+ * Otherwise this function is just the identity
+ *
+ * @param {unknown} value
+ * @returns {unknown}
+ */
+const styleObject = value => {
+  if (value && getPrototypeOf(value) === null) {
+    return fromEntries(entries(value).map(([k, v]) => [k, styleObject(v)]));
+  }
+  return value;
+};
+
+/**
+ * Prepends a `CompartmentDescriptor.path` with the computed canonical name (or "[Root]" if entry compartment).
+ *
+ * Returns a shallow copy of `path` with a {@link CustomInspectFunction}.
+ *
+ * @param {string[]} path
+ */
+const stylePath = path => {
+  const fancyPath = [...path];
+  assign(fancyPath, {
+    /** @type {CustomInspectFunction} */
+    [inspect.custom]: (_, { stylize }) =>
+      path.length
+        ? `[${stylize('Canonical', 'date')}: ${stylize(path.join('>'), 'special')}] ${inspect(path)}`
+        : `[${stylize('Root', 'date')}]`,
+  });
+  return fancyPath;
+};
+
+/**
+ * Dump a {@link CompartmentMapDescriptor}, omitting some fields.
+ *
+ * - {@link CompartmentMapDescriptor.tags} is omitted
+ * - A `CompartmentDescriptor` will only show the following fields: `label`,
+ *   `name`, `scopes`, `location`, `sourceDirname`, `modules`, and `path`
+ *
+ * This can be used on any `CompartmentMapDescriptor` and is not purpose-built
+ * for `ProjectFixture` objects.
+ *
+ * @template [Context=unknown]
+ * @param {{log: LogFn}|ExecutionContext<Context>} logger
+ * @param {CompartmentMapDescriptor} compartmentMap
+ * @returns {void}
+ */
+export const dumpCompartmentMap = (logger, compartmentMap) => {
+  const compartmentMapForInspect = {
+    ...compartmentMap,
+    /** @type {CustomInspectFunction} */
+    [inspect.custom]: () => ({
```

Comment:
I'd consider embedding this stuff into the "real" data structure (or even creating a `CompartmentDescriptor` class) but since this is a node-ism we'd need to wrap it in a power. Not worth it.